### PR TITLE
CARTO module: Integrate Maps API v2 endpoints

### PR DESCRIPTION
**Pull request by alasarr**
Include support for the new endpoints of Maps API v2.

### CODE REVIEW

File: test/modules/carto/mock-fetch.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* global global, window */
+
+export function mockTileJSON() {
+  const _global = typeof global !== 'undefined' ? global : window;
+  _global.fetch = url =>
```
Comment (Pessimistress) [inline test/modules/carto/mock-fetch.js:5]: You need to intercept `fetch` and restore after each test, or this will break other layers.

Comment (alasarr) [reply test/modules/carto/mock-fetch.js:5]: On it, thanks

Comment (alasarr) [reply test/modules/carto/mock-fetch.js:5]: Fixed and I've finally fixed my local dev, I can run browser tests there 🥳

File: modules/carto/src/api/maps-api-client.js [hunk 2]
```diff
@@ -3,32 +3,45 @@ import {getDefaultCredentials, getMapsVersion} from '../config';
...
 }
 
 /**
- * Instantiate a map using Maps API
+ * Request against Maps API
  */
-async function instantiateMap({mapConfig, credentials}) {
-  const url = buildURL({mapConfig, credentials});
-
+async function request({url, credentials}) {
```
Comment (AdriSolid) [inline modules/carto/src/api/maps-api-client.js:44]: Function name proposals:
`createRequest`
`makeRequest`

File: modules/carto/src/config.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ export function getDefaultCredentials() {
 }
 
 export function getMapsVersion(creds) {
-  const localCreds = creds || credentials;
+  const localCreds = {...credentials, ...creds};
   if (localCreds.mapsVersion) {
     return localCreds.mapsVersion;
   }
```
Comment (AdriSolid) [inline modules/carto/src/config.js:30]: I would apply destructuring here
```javascript
export function getMapsVersion(creds) {
  const {mapsVersion, mapsUrl} = {...credentials, ...creds};

  if (mapsVersion) {
    return mapsVersion;
  }

  return mapsUrl.includes('api/v1/map') ? 'v1' : 'v2';
}

**Review by AdriSolid — CHANGES_REQUESTED** [review]
:ok_hand:

**Comment by alasarr** [comment]
I still need to fix the coverage, it has been decreased too much

**Review by AdriSolid — APPROVED** [review]

**Comment by alasarr** [comment]
I'm going to create another PR with a test refactor, I want to keep it separated to reduce the size of this

**Pull request merged**